pinecone/client.go

```diff
@@ -132,6 +400,62 @@ func (c *Client) ListIndexes(ctx context.Context) ([]*Index, error) {
 	return indexes, nil
 }
 
+// CreatePodIndexRequest holds the parameters for creating a new pods-based index.
+//
+// Fields:
+//   - Name: The name of the pods-based index to be created.
+//   - Dimension: The dimension of the index (must match [dimensionality] of upserted vectors).
+//   - Metric: The metric used to measure the [similarity] between vectors.
+//   - Environment: The [cloud environment] in which the index will be created.
+//   - PodType: The [type of pod] to use for the index.
+//   - Shards: The number of shards to use for the index (defaults to 1).
+//   - Replicas: The number of [replicas] to use for the index (defaults to 1).
```

Comment:
Along with my previous comments, I'd take a look at how we represent some of these fields in the OAS documentation and lift that, just for specificity:

- `replicas`: https://github.com/pinecone-io/apis/blob/fbd9d8d48c95b12c6b28aea4e496b6f28666d2d1/src/release/control/resources/indexes/models/PodSpecReplicas.yaml#L3
- `shards`: https://github.com/pinecone-io/apis/blob/fbd9d8d48c95b12c6b28aea4e496b6f28666d2d1/src/release/control/resources/indexes/models/PodSpecShards.yaml#L3

etc